Develop branch adapts to openharmony

qiuguohua opened the pull request:
Re: #

### Changelog
Adaptation modules.
1、File
2、Audio
3、Network
4、Project
5、Font
etc.

-------

### Continuous Integration

This pull request:

* [x] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

<!-- Note: Makes sure these boxes are checked before submitting your PR - thank you!
- [ ] Your pull request title is using English, it's precise and appropriate.
- [ ] If your pull request has gone "stale", you should **rebase** your work on top of the latest version of the upstream branch.
- [ ] If your commit history is full of small, unimportant commits (such as "fix pep8" or "update tests"), **squash** your commits down to a few, or one, discreet changesets before submitting a pull request.
- [ ] Document new code with comments in source code based on API docs
- [ ] Make sure any runtime log information in `log` , `error` or `new Error('')` has been moved into `EngineErrorMap.md` with an ID, and use `logID(id)` or `new Error(getError(id))` instead.
- To official teams:
  - [ ] Check that your PR is following our [guides](https://github.com/cocos/3d-tasks/blob/master/workflows/readme.md)
-->

qiuguohua committed:
OpenHarmony adapts 3.6.1-oh (#12780)

cocos-robot committed:
[ci skip][AUTO]: Automated code generating update: 641b340d1672f098516ca7110b0ac2cbadb4b638  (#12780) (#12840)

Co-authored-by: cocos-robot <cocos-robot@cocos.com>

oahcy committed:
openharmony not support Expand syntax， update external （add libwebp，physx support）。 (#12845)

* update external tag

* fix : oh not support ...args

oahcy committed:
update external (#12856)

qiuguohua committed:
Fixing the memory leak is only partially solved, and the OpenHarmony system still needs to be solved (#12872)

qiuguohua committed:
Fixed an issue where arguments with indefinite parameters could not be used (#12875)

oahcy committed:
config npm when build openharmony proj (#12876)

* config npm when build openharmony proj

qiuguohua committed:
Change the application to a singleton (#12883)

oahcy committed:
V3.6.1 oh arm64 support (#12994)

* openharmony arm64 support

* update exteral config

cocos-robot committed:
[ci skip][AUTO]: Automated code generating update: fd2158800f594ccac7ec2eac463ede27445e793b  (#12994) (#13000)

Co-authored-by: cocos-robot <cocos-robot@cocos.com>

oahcy committed:
Oh network adapter (#13013)

oahcy committed:
openharmony abi setting support (#13012)

oahcy committed:
openharmony support download (#13033)

oahcy committed:
update native/external-config.json (#13036)

oahcy committed:
V3.6.1 oh network (#13052)

* openharmony network support

* Revert "Oh network adapter (#13013)"

This reverts commit 6a3849f137b9bcf183e138edf9139f9cf071a4e0.

oahcy committed:
update external config (#13059)

qiuguohua committed:
openharmony application switch from FA model to stage model (#13069)

* openharmony application switch from FA model to stage model
https://docs.openharmony.cn/pages/v3.2Beta/zh-cn/application-dev/ability/ability-brief.md/

* Removal of redundant logs.
Add 64-bit support

* Parsing json with JSON5

oahcy committed:
openharmony dir add to editor pack (#13081)

* add openharmony for win and mac

* update external json

qiuguohua committed:
openharmony adapts editbox (#13300)

* openharmony adapts editbox, and there is another problem is that the editbox dialog box pushes the game page up

* 1.Change 'callback' to 'function'；

* Change 'IMEDialog' to 'EditBoxDialog'

* Add the copyright of the template

* Add openharmony editbox-related interface definitions to editbox-openharmony.h

oahcy committed:
update external config (#13355)

qiuguohua committed:
OpenHarmony Device Interface Adaptation (#13307)

* OpenHarmony Device Interface Adaptation

1. Remove useless files.
2. modification of copyright information.
3. Add security judgment

qiuguohua committed:
Fix the problem that when music is played with AudioSource and suddenly quit, it is still playing. (#13408)

* Fix the problem that when music is played with AudioSource and suddenly quit, it is still playing.

* Fix eslint error

qiuguohua committed:
V3.6.1 openharmony adapts webview (#13797)

* openharmony adapts webview functionality

* 1. Fix the problem that the gyroscope does not work
2. webview adaptation optimization.
3. editbox adaptation optimization

* Adding missing headers

* Add English comment

qiuguohua committed:
openharmony adaptation video (#13876)

PPpro committed:
remove interface check workflow (#14283)

qiuguohua committed:
Merge remote-tracking branch 'origin/oh' into develop-oh-merge1

# Conflicts:
#	native/cocos/audio/android/PcmAudioService.h
#	native/cocos/audio/android/audio.h
#	native/cocos/audio/android/utils/Utils.cpp
#	native/cocos/audio/common/utils/include/primitives.h
#	native/cocos/audio/common/utils/include/tinysndfile.h
#	native/cocos/base/TypeDef.h
#	native/cocos/bindings/auto/jsb_cocos_auto.cpp
#	native/cocos/bindings/auto/jsb_cocos_auto.h
#	native/cocos/bindings/manual/jsb_module_register.cpp
#	native/cocos/engine/Engine.cpp
#	native/external-config.json
#	pal/env/native/env.ts
#	pal/screen-adapter/native/screen-adapter.ts
#	platforms/native/builtin/index.js
#	platforms/native/engine/jsb-gfx.js
#	platforms/native/engine/jsb-loader.js
#	scripts/build-engine/package-lock.json
#	scripts/build-engine/package.json
#	scripts/build-engine/src/index.ts

qiuguohua committed:
Upgrade openharmony to develop branch

qiuguohua committed:
Restore the already supported parts of openharmony

qiuguohua committed:
1.Fix the screen adaptation problem
2.openharmony removes the download interface

qiuguohua committed:
Update copyright information

qiuguohua commented:
@cocos-robot run test cases

minggo commented on native/cocos/bindings/manual/jsb_global.cpp:
```diff
@@ -804,7 +808,7 @@ static bool JSB_setPreferredFramesPerSecond(se::State &s) { // NOLINT
 SE_BIND_FUNC(JSB_setPreferredFramesPerSecond)
 
-#if CC_USE_EDITBOX
+#if CC_USE_EDITBOX 
```
extra space

qiuguohua replied:
OK

minggo commented on scripts/native-pack-tool/package.json:
```diff
@@ -17,6 +17,7 @@
     "yargs": "^17.5.1",
-    "zlib": "^1.0.5"
+    "zlib": "^1.0.5",
+    "json5":"^2.2.1"
```
@PPpro please help to review it too.

PPpro replied:
it's OK to add json5 dependency

minggo commented on templates/openharmony/AppScope/resources/base/element/string.json:
```diff
@@ -0,0 +1,8 @@
+  "string": [
+    {
+      "name": "app_name",
+      "value": "OpenHarmonyDemo"
```
OpenHarmonyDemo -> OpenHarmonyApp

qiuguohua replied:
OK

minggo commented on templates/openharmony/package.json:
```diff
@@ -0,0 +1,21 @@
+{
+    "license": "ISC",
```
What's the license?

qiuguohua replied:
This is the default license generated by DecEco when creating a new project

qiuguohua committed:
Update version number

qiuguohua committed:
Delete extra spaces

dumganhar commented on native/cocos/bindings/manual/jsb_conversions.h:
```diff
@@ -1230,8 +1230,6 @@ inline bool nativevalue_to_se(const ccstd::vector<T> &from, se::Value &to, se::O
             auto *pFrom = ccnew T(from[i]);
             nativevalue_to_se(pFrom, tmp, ctx);
-            tmp.toObject()->clearPrivateData();
-            tmp.toObject()->setPrivateData(pFrom);
```
Do these two lines cause any issues?

qiuguohua replied:
Not sure, there should be a problem before

dumganhar commented on native/cocos/network/HttpRequest.h:
```diff
@@ -241,8 +241,8 @@ class CC_DLL HttpRequest : public RefCounted {
-    inline float getTimeout() const {
-        return _timeoutInSeconds;
+    inline long getTimeout() const {
+        return static_cast<long>(_timeoutInSeconds);
```
Why change the type of return value, which will break the compatibility?
_timeoutInSeconds could be a float value, for instance, 1.5f means 1.5 seconds.

qiuguohua replied:
I'll try

PPpro commented on cocos/serialization/deserialize-dynamic.ts:
```diff
@@ -253,7 +253,7 @@ function compileDeserializeNative (_self: _Deserializer, klass: CCClassConstruct
                 continue;
             }
-            if (!fastMode && typeof prop !== 'object') {
+            if (typeof prop !== 'object') {
```
why remove this fastMode

qiuguohua replied:
There is a problem with not removing this serializatio

dumganhar replied:
What problem?

dumganhar commented on native/cocos/platform/interfaces/modules/Device.cpp:
```diff
@@ -30,9 +30,32 @@
+}
+
+uint32_t Device::getInnerHeight() {
+    ISystemWindow* systemWindow = CC_GET_MAIN_SYSTEM_WINDOW();
```
ISystemWindow was bound, could we get ISystemWindow in JS directly?

qiuguohua replied:
Only the interface is exported, not the instance? This instance cannot be created directly.

dumganhar replied:
How about using `root.mainWindow.width/height`?

qiuguohua replied:
If we can use this interface, can't window.innerWidth be deleted?

dumganhar replied:
Should not delete `window.innerWidth` which is for keeping compatibility with `web`.
So where do you use `Device::getInnerHeight` after you exported `Device::getInnerHeight` to JS?

qiuguohua committed:
1.Restore the 'fastmode' judgment
2.Replace ‘device.getInnerWidth’ interface with ‘cclegacy.Director.root.mainWindow.width’.

qiuguohua committed:
'root.mainWindow.width'' is undefined, use 'getinner'' interface
